sdk/src/store.rs

```diff
@@ -2080,16 +2082,24 @@ impl Store {
             let label = crate::jumbf::labels::assertion_label_from_uri(&uri.url())
                 .ok_or(Error::BadParam("write_dynamic_assertions".to_string()))?;
 
-            let da_size = da.reserve_size();
+            let da_size = da.reserve_size()?;
             let da_data = if _sync {
                 da.content(&label, Some(da_size), preliminary_claim)?
             } else {
                 da.content(&label, Some(da_size), preliminary_claim).await?
             };
 
-            // TO DO: Add new assertion to preliminary_claim
-            // todo: support for non-CBOR asssertions?
-            final_assertions.push(UserCbor::new(&label, da_data).to_assertion()?);
+            match da_data {
+                DynamicAssertionContent::Cbor(data) => {
+                    final_assertions.push(UserCbor::new(&label, data).to_assertion()?);
+                }
+                DynamicAssertionContent::Json(data) => {
+                    final_assertions.push(User::new(&label, &data).to_assertion()?);
+                }
+                DynamicAssertionContent::Binary(format, data) => {
+                    // todo:: support for pushing binary assertions here
```

Comment:
Suggestion: Add a `todo!(...)` macro here so we don't silently fail in the binary case.

Reply:
done!